python/idsse_common/idsse/common/utils.py

```diff
@@ -206,3 +209,43 @@ def round_half_away(number: float, precision: int = 0) -> Union[int, float]:
         else _round_away_from_zero(factored_number)
     ) / factor
     return int(rounded_number) if precision == 0 else float(rounded_number)
+
+def shrink_grib(filename: str, variables: list[str]) -> None:
+    """
+    Shrink a grib file extracting only those variables from the list provided.
+    Notes:
+        1. Variable names may not be unique so several variables with the same
+        name may appear in the result file, e.g. deterministic and probabilistic
+        versions of the same name
+        2. Variables not found in the file are ignored
+        3. The pygrib package when updated may change parameter (variable) names
+        for a given product users should be aware and maintain product mapping names
+        appropriately.
+
+    Args:
+        filename (str): the grib file to be reduced
+        variables: the list af pygrib variable names to be retained
+    Returns:
+        None
+    """
+    if not variables:
+        return  # Nothing to do!!!
+    # Use a set for faster lookups...
+    req_vars = set(variables)
+    try:
+        # Open a work file for the result..
+        grb_out = open(f'{filename}{".tmp"}', 'wb')
```

Comment:
Mackenzie comment about f-string is true for about for lines of code, this being the first in this file. 
Here this line should be: grb_out = open(f'{filename}.tmp', 'wb').

The second thing I see is that the file isn't being opened with 'with', so if an exception is thrown the file isn't closed.

Reply:
Used "with"